lib/temporal/client/retryer.rb

```diff
@@ -0,0 +1,29 @@
+module Temporal
+  module Client
+    module Retryer
+      INITIAL_INTERVAL_S = 0.2
+      MAX_INTERVAL_S = 6.0
+      BACKOFF_COEFFICIENT = 1.2
+
+      # Used for backoff retries in certain cases when calling temporal server.
+      # metadata_hash: for logging, pass in metadata.to_h
+      def self.retry_for(give_up_after_s, retry_message:, metadata_hash:, &block)
+        # Values taken from the Java SDK
+        # https://github.com/temporalio/sdk-java/blob/ad8831d4a4d9d257baf3482ab49f1aa681895c0e/temporal-serviceclient/src/main/java/io/temporal/serviceclient/RpcRetryOptions.java#L32
+        current_interval_s = INITIAL_INTERVAL_S
+        elapsed_s = 0.0
+        loop do
+          begin
+            return yield
+          rescue
```

Comment:
We most likely don't want a blanket `rescue` here, this might be dangerous since it can mask things like `SyntaxError` or `NoMemoryError`

Reply:
I believe this should be fine. Without a type, rescue catches `StandardError` which does not include `NoMemoryError`, `SyntaxError` (a subclass of `ScriptError`) and some others.

Reply:
Ah, you're right, all good 👍